### DISCUSSION TITLE
Contribution and possible merge of my project

### DISCUSSION THREAD

**Discussion by ZigRazor** [stdgraph/graph-v2#32] [Show and tell]
Hi all, in these year I start a similar project for an header only graph library.
You can see it at https://github.com/ZigRazor/CXXGraph .

My project have a good set of tests and also benchmark on it,
Is C++17 ready,
A good CI/CD framework on Github other than a lot of algorithms already implemented and optimized over the structures.

What do you think about it? 
You think is possible to merge the 2 works to do a good unique product?

I'm open to discussion @neoblizz @pratzl @lums658

**Comment by pratzl** [comment]
Talking about merging seems a little aggressive, largely because we've used C++ features like ranges, views and concepts that can dramatically offer new opportunities and change the way you approach a problem. If you'd like to contribute, I noticed that you've written a number of algorithms which is an area that we're weak right now. What do you think about looking at one of the algorithms in our README and porting any of your algorithms that's on the list?

**Comment by ZigRazor** [comment]
Hi @pratzl, 
I understand your speech, but I don't won't to leave my project, where I work for years, and work for another one without any credits of all the work done.
I think you have had a good idea to propose your project as an introduction in the standard ( it is something that miss in the actual standard, and with the so many applications of Graph I think the introduction in the STL is necessary ), but throw away all the work done in this year to collect and implement algorithms is very difficult for me, also because the community that I have create around my project can be very useful also for this project.
I will think about your invitation, but I think that a sort of merge also of the "Know How" is better.

**Comment by pratzl** [comment]
Understood, and I appreciate where you're coming from.

Things are changing with us also where there's a desire to keep the contributors to a tight group, so I think there's a mutual feeling that this isn't a good time right now.

I suggest we keep an open mind for the future, and good luck in your effort!


Sent from Outlook<http://aka.ms/weboutlook>
